opm/input/eclipse/Schedule/Well/WellPropertiesKeywordHandlers.cpp

```diff
@@ -500,7 +500,7 @@ void handleWTEMP(HandlerContext& handlerContext)
 
         for (const auto& well_name : well_names) {
             const auto& well = handlerContext.state().wells.get(well_name);
-            const double current_temp = !well.isProducer()? well.temperature(): 0.0;
+            const double current_temp = well.hasTemperature()? well.temperature(): 0.0;
```

Comment:
since we changes to use std::optional for the well (injection) temperature, should we make sure it is std::nullopt for producers?